scripts/pre_commit.sh

```diff
@@ -1,15 +1,16 @@
 #!/bin/bash
 set -e
 
-./scripts/build_proto.sh
+# start tests in parallel (with additional parallelism)
+pytest -m fast -n auto &
+
+# check bandit in parallel
+bandit -r src -ll &
+
+# run API documentation test notebooks in parallel
+./scripts/nb_test.sh && pytest examples/api --cov-fail-under 0 &
 
 # fix isort and format with black
-isort .
-black src tests
-pre-commit run --all-files
-bandit -r src -ll
-pytest -m fast
-
-# run API documentation test notebooks
-./scripts/nb_test.sh
-pytest examples/api --cov-fail-under 0
\ No newline at end of file
+./scripts/build_proto.sh && isort . && black src tests && pre-commit run --all-files &
```

Comment:
Yes, and so far it takes my run time from 30 seconds down to 17, without any issue, but if there are problems we can go back to the old way, I just wanted to try to get that number down so that people are more inclined to run it more frequently.

Reply:
I am keen to hear if it works for other people since I haven't been able to test it on any other setups like Linux. I assume the set -e should still trigger a failure if one of the backgrounded tasks fails but maybe im wrong.